### PR TITLE
support strings in `elements.get_node_id()`

### LINKED ISSUE #519: BUG: get_node_id does not support string IDs
If `edge_id` column is using string IDs, this line will fail as numpy supports only numeric dtypes.

https://github.com/pysal/momepy/blob/78c1b244f3acef60d77cbce98e3426c755b3033d/momepy/elements.py#L807C16-L807C24

We need to change it to `pd.isna`.

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR resolves #519

### CODE REVIEW

**Comment by jGaboardi** [comment]
Seeing now that the line isn't being hit by any tests...

**Comment by martinfleis** [comment]
It is technically covering a corner case that should not happen when everything is as expected.

We hit it when we don't have a street network edge attached to a building footprint, which happens only if a building is really far away from any street. But we could possibly simulate that by using `momepy.get_network_id` with a small `min_size` (that API and implementation there are bad...)

**Comment by jGaboardi** [comment]
> It is technically covering a corner case that should not happen when everything is as expected.
> 
> We hit it when we don't have a street network edge attached to a building footprint, which happens only if a building is really far away from any street. But we could possibly simulate that by using `momepy.get_network_id` with a small `min_size` (that API and implementation there are bad...)

Working on creating a synthetic test case now that simply converts some edge IDs in the `bubenec` streets to NaN.

**Comment by jGaboardi** [comment]
Passing locally and hitting the offending line.

File: momepy/tests/test_elements.py [hunk 1]
```diff
@@ -154,8 +155,16 @@ def test_get_node_id(self):
...
             self.df_buildings, self.df_streets, "nID"
         )
-        ids = mm.get_node_id(self.df_buildings, nodes, edges, "nodeID", "nID")
-        assert not ids.isna().any()
+        ids1 = mm.get_node_id(self.df_buildings, nodes, edges, "nodeID", "nID")
+        assert not ids1.isna().any()
+
+        # test for NaNs within `object` nIDs column
+        _df_buildings = self.df_buildings.copy()
+        _df_buildings.loc[[0, 1], "nID"] = pd.NA
```
Comment (martinfleis) [inline momepy/tests/test_elements.py:163]:
```suggestion
        _df_buildings["nID"] = _df_buildings["nID"].astype(str)
        _df_buildings.loc[[0, 1], "nID"] = pd.NA
```

Might be worth actually testing for the string ID?

Comment (jGaboardi) [reply momepy/tests/test_elements.py:163]: Sure. Will push up a new version.

**Review by martinfleis — APPROVED** [review]
Thanks!
